Навёл красоты

danilasar opened the pull request:
Метаданные подсасываются из details.toml, добавил функцию леммы и окончание доказательства.
![Screenshot From 2025-06-10 11-30-54](https://github.com/user-attachments/assets/3e102651-f286-41c2-a682-010fe9d37bd8)

Именование файлов секций наше стандартное, `sections/{номер}.typ`. Секции перечисляются массивом в `sections.sectons` файла `details.typ`. При необходимости можно управлять нумерацией билетов прямо из конфига (вкл/выкл), перед номером можно написать префикс ("Билет","Блок", "Вопрос", ...).

danilasar committed:
feat: configuration in details.toml

danilasar committed:
feat: useful functions & sections generator

danilasar committed:
refactor: decreased files number

danilasar committed:
fix: garbage and shit in main.typ removed

danilasar committed:
fix: some fixes

aragami3070 commented:
Либо я не разобрался, либо оно немного через очко работает
![изображение](https://github.com/user-attachments/assets/d5ea87da-0481-437b-add3-f4e90b8faf69)

я всего лишь добавил 
= Какой-то текст в 01.typ

aragami3070 commented:
Ладно, я понял как оно работает. Но на это надо Readme 100%, вообще не интуитвное

aragami3070 commented on main.typ:
```diff
@@ -1,8 +1,21 @@
+  i += 1
+}
+
 #pagebreak()
```
pagebreak лишний, его в конфиг надо засунуть на каждую секцию

aragami3070 replied:
На каждую, кроме последней

aragami3070 commented on main.typ:
```diff
@@ -1,8 +1,21 @@
+    }
+    numbering += str(i) + ". "
+  }
+  heading(section, numbering: numbering)
```
При попытке добавить файл 02.typ и заголовок в toml файл в массив sections, оно орет на меня вот этим:
```
error: invalid numbering pattern
   ┌─ main.typ:16:30
   │
16 │   heading(section, numbering: numbering)
   │                               ^^^^^^^^^
```

aragami3070 reviewed (changes requested):
- Надо подробное Readme по тому как этим пользоваться, чтобы каждому лично не объяснять это
- Фиксы, которые я пометил
- Тот error на numbering блочит возможность сделать больше 1 section

danilasar committed:
fix!: par inside paragraph
BREAKING CHANGE: In Typst 0.13 were accepted distinction between
proper paragraphs and just inline-level content. My old
implementation for proofable_block is not valid anymore
because there is a situation where paragraph becomes a
part of other paragraph.

danilasar commented:
Коммит 6132eb2 невероятно важен. В Typst 0.13 было введено чёткое разграничение между параграфами и блоками, поэтому старая реализация block_proofable ныне невалидна. Я исправил block_base, чтобы обеспечить совместимость с новой логикой языка.

danilasar commented:
Спасибо @mickleon за сообщение о странном баге в функции теоремы, он был связан как раз с этим обновленем

danilasar committed:
fix: numbering for multiple sections, pagebreak

danilasar commented:
> Надо подробное Readme по тому как этим пользоваться, чтобы каждому лично не объяснять это

Писать это в README.md, возможно, плохая затея, потому что он будет копироваться вместе со всем остальным при создании репозитория на основе шаблона. Каждый раз удалять его будет весьма запарно.

У гитхаба есть викистраницы, я предлагаю воспользоваться ими.

danilasar committed:
fix: a bit bug in section numbering

danilasar commented:
> Либо я не разобрался, либо оно немного через очко работает ![изображение](https://private-user-images.githubusercontent.com/163641422/453368146-d5ea87da-0481-437b-add3-f4e90b8faf69.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDk2NTY2NzcsIm5iZiI6MTc0OTY1NjM3NywicGF0aCI6Ii8xNjM2NDE0MjIvNDUzMzY4MTQ2LWQ1ZWE4N2RhLTA0ODEtNDM3Yi1hZGQzLWY0ZTkwYjhmYWY2OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNjExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDYxMVQxNTM5MzdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jMmQ4NjAxOGI2YTVlOTdlODIyMmVlZTAxZjMyMWY0ZGNiMjg5NzNiMjZmMWY1OWEzYTVkMTJmNzAyNDlmMGM1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.aSO5uA8oMr7aw2taY9wA2JjXVP0fBS3wMh4eWwjrWro)
> 
> я всего лишь добавил = Какой-то текст в 01.typ

Ну так всё логично, это не новый вопрос, а заголовок первого уровня. Можно, конечно, для содержимого секций сделать смещение уровней, чтобы = автоматически переделывался в ==, это не сложно, просто я решил, что такое наоборот собьёт с толку.

aragami3070 commented:
> > Надо подробное Readme по тому как этим пользоваться, чтобы каждому лично не объяснять это
> 
> Писать это в README.md, возможно, плохая затея, потому что он будет копироваться вместе со всем остальным при создании репозитория на основе шаблона. Каждый раз удалять его будет весьма запарно.
> 
> У гитхаба есть викистраницы, я предлагаю воспользоваться ими.

Ну тогда есть какой-то Contributions.md у гитхаба, туда напиши, как работать с секциями

aragami3070 reviewed (approved):
Напиши Contributions.md или как он там, как написал выше. В остальном LG TV

danilasar commented:
https://github.com/SE-legacy/exam-manual-typst-template/wiki

З. Ы. Указание на вики оставил в Description к репозиторию. Если будут возникать вопросы, так уж и быть, добавим README.md. CONTRIBUTING.md &mdash; это другое, тем более он тоже наследуется
